lib/hashtable.cc

```diff
@@ -257,72 +260,77 @@ void Hashtable::get_kmers(const std::string &s,
 void Hashtable::get_kmer_hashes(const std::string &s,
                                 std::vector<HashIntoType> &kmers_vec) const
 {
-    KmerIterator kmers(s.c_str(), _ksize);
+    KmerHashIterator * kmers = new_kmer_iterator(s);
 
-    while(!kmers.done()) {
-        HashIntoType kmer = kmers.next();
+    while(!kmers->done()) {
+        HashIntoType kmer = kmers->next();
         kmers_vec.push_back(kmer);
     }
+    delete kmers;
 }
 
 
 void Hashtable::get_kmer_hashes_as_hashset(const std::string &s,
         SeenSet& hashes) const
 {
-    KmerIterator kmers(s.c_str(), _ksize);
+    KmerHashIterator * kmers = new_kmer_iterator(s);
 
-    while(!kmers.done()) {
-        HashIntoType kmer = kmers.next();
+    while(!kmers->done()) {
+        HashIntoType kmer = kmers->next();
         hashes.insert(kmer);
     }
+    delete kmers;
 }
 
 
 void Hashtable::get_kmer_counts(const std::string &s,
                                 std::vector<BoundedCounterType> &counts) const
 {
-    KmerIterator kmers(s.c_str(), _ksize);
+    KmerHashIterator * kmers = new_kmer_iterator(s);
```

Comment:
I would wrap it in `std::unique_ptr< KmerHashIterator > kmers(new_kmer_iterator(s))` and then get rid of the `delete`.